lang/cs/Org.Apache.REEF.Network/Elastic/Comm/Impl/TopologyMessagePayload.cs

```diff
@@ -0,0 +1,135 @@
+﻿// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+using Org.Apache.REEF.Tang.Exceptions;
+using Org.Apache.REEF.Utilities;
+using Org.Apache.REEF.Utilities.Attributes;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace Org.Apache.REEF.Network.Elastic.Comm.Impl
+{
+    /// <summary>
+    /// Messages sent by the driver to operators. 
+    /// This message contains information for the destination node on the topology.
+    /// </summary>
+    [Unstable("0.16", "API may change")]
+    internal class TopologyMessagePayload : DriverMessagePayload
+    {
+        /// <summary>
+        /// Create a driver message payload containing topology updates.
+        /// </summary>
+        /// <param name="updates">The topology updates</param>
+        /// <param name="toRemove">Whether the updates are additions to the current topology state or nodes removal</param>
+        /// <param name="stageName">The stage context for the message</param>
+        /// <param name="operatorId">The id of the operator receiving the topology update</param>
+        /// <param name="iteration">The iteration in which the update takes effect</param>
+        public TopologyMessagePayload(DriverMessagePayloadType type, List<TopologyUpdate> updates, string stageName, int operatorId, int iteration)
+            : base(stageName, operatorId, iteration)
+        {
+            PayloadType = type;
+            TopologyUpdates = updates;
+        }
+
+        /// <summary>
+        /// Clone the message.
+        /// </summary>
+        /// <returns>An object containing the shallow copy of the message.</returns>
+        public override object Clone()
+        {
+            var updatesClone = new List<TopologyUpdate>();
+
+            foreach (var update in TopologyUpdates)
+            {
+                var clone = new TopologyUpdate(update.Node, update.Children, update.Root);
+                updatesClone.Add(update);
+            }
```

Comment:
in Linq, that's 
```C#
var updatesClone = TopologyUpdates.Select(
    update => new TopologyUpdate(update.Node, update.Children, update.Root)); // .ToList();
```
(I would also consider using more generic container types, like `IEnumerable` whenever possible)

(and, maybe, implement `ICloneable` in `TopologyUpdate`?)

Reply:
Fixed.